### DISCUSSION TITLE
Telemetry

### DISCUSSION THREAD

**Discussion by arokem** [dipy/dipy#3568] [1. Ideas & Feedback]
We'd like to understand who is using the software and get some data about DIPY usage. This is valuable not only so that we can plan our developments (for example, how many people are using the software on different OS), but also in justifying our existence to funders and other stakeholders.

One relatively easy way to do this is to use [popylar](https://popylar.github.io/), which records an import of a software library using Google Analytics. This kind of approach is already used by other projects (for example) by the homebrew package managed (see: https://docs.brew.sh/Analytics).

Another approach, used in fmriprep uses [sentry](https://sentry.io) to record usage and also report typical errors. See: https://github.com/poldracklab/fmriprep/blob/master/fmriprep/utils/sentry.py

I still need to look more at this latter approach. 

My proposal: let's progress slowly. At first, we can implement a `dipy.telemetry` module that contains the popylar/sentry code. When this module is imported, this event is recorded with GA/Sentry. So initially, users would have to integrate this into their own code. Workflows would have a `--telemetry` flag that would have to be included for the usage to be recorded.

This means that users would have to *opt in* every time they want their use of the software to be recorded. And means that we will probably get very little data initially.

Either way, I think we should do this on 1.1, after 1.0 is out. 

How do people feel about this?

**Comment by arokem** [comment]
@oesteban : could you please tell us a bit about your experience with Sentry?

**Comment by satra** [comment]
@arokem - we started building a service here: https://github.com/mgxd/etelemetry

it's quiet simple a service to provide software version updates.

**Comment by oesteban** [comment]
@arokem sometimes the amount of feedback can be overwhelming, depending on your user base. If you use it to trace almost everything, they you easily get throttled with the free tier.

If you just record when things start and finish then it is potentially a good option. The API could be better documented, basic things are pretty trivial but it becomes hard when you want better annotated traces.
